lucene/core/src/java/org/apache/lucene/search/LeafFieldComparator.java

```diff
@@ -116,4 +116,12 @@
    * obtain the current hit's score, if necessary. */
   void setScorer(Scorable scorer) throws IOException;
 
+  /** Publishes feature values for the given docID
+   *  The passed in docID values should only move ahead --
+   *  DocValues do not support reverse iterations.
+   *  WARNING: Can be a costly API, especially for sorted fields
+   *
+   *  @param doc docID relative to current reader
+   */
+  T getDocValue(int doc) throws IOException;
```

Comment:
I don't understand why this is needed, if you know the bottom slot you can simply call `FieldComparator::value(int)` ?

Reply:
`comparator` here is a `LeafFieldComparator`, and we need the feature value for a given docID, which might not necessarily be the bottom value.